### PR TITLE
Added support for custom CIDRs for bottlerocket

### PR DESCRIPTION
**Pull request by edibble21**
Fixes #N/A 

**Description**
Added support for custom CIDRs for bottlerocket
**How was this change tested?**
make presubmit
**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/providers/launchtemplate/suite_test.go [hunk 1]
```diff
@@ -1426,6 +1426,8 @@ essential = true
 			BeforeEach(func() {
 				nodeClass.Spec.AMISelectorTerms = []v1.AMISelectorTerm{{Alias: "bottlerocket@latest"}}
 				nodeClass.Spec.Kubelet = &v1.KubeletConfiguration{MaxPods: lo.ToPtr[int32](110)}
+
+				awsEnv.LaunchTemplateProvider.ClusterCIDR.Store(nil)
```
Comment (njtran) [inline pkg/providers/launchtemplate/suite_test.go:1430]: Do we need to add a test that checks the user data that we created?

File: pkg/providers/amifamily/bootstrap/bottlerocket.go [hunk 2]
```diff
@@ -95,3 +99,11 @@ func (b Bottlerocket) Script() (string, error) {
 	}
 	return base64.StdEncoding.EncodeToString(script), nil
 }
+
+func GetDNSIPFromCIDR(cidr *string) (*string, error) {
```
Comment (rschalo) [inline pkg/providers/amifamily/bootstrap/bottlerocket.go:103]: Is this exported?

File: pkg/providers/amifamily/bootstrap/bottlerocket.go [hunk 3]
```diff
@@ -51,7 +55,7 @@ func (b Bottlerocket) Script() (string, error) {
 	}
 
 	if b.KubeletConfig != nil {
-		if len(b.KubeletConfig.ClusterDNS) > 0 {
+		if len(b.KubeletConfig.ClusterDNS) > 0 && b.ClusterCIDR == nil {
```
Comment (engedaam) [inline pkg/providers/amifamily/bootstrap/bottlerocket.go:58]: Just for my learning, why we need to update this check to consider cluster CIDR?

Comment (edibble21) [reply pkg/providers/amifamily/bootstrap/bottlerocket.go:58]: So you're correct I did not make this the most clear. If I am understanding correctly, in the event you don't have a custom CIDR to get the ClusterDNSIP you use the the default DNSIP from the KubeletConfig. So now that it supports custom CIDRs we would need to check for that. If that does not make sense let me know

**Pull request closed without merging**